src/main/java/io/kestra/plugin/ai/tool/Agent.java

```diff
@@ -0,0 +1,215 @@
+import dev.langchain4j.exception.ToolExecutionException;
+import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
+import dev.langchain4j.rag.DefaultRetrievalAugmentor;
+import dev.langchain4j.rag.content.retriever.ContentRetriever;
+import dev.langchain4j.rag.query.router.DefaultQueryRouter;
+import dev.langchain4j.rag.query.router.QueryRouter;
+import dev.langchain4j.service.AiServices;
+import dev.langchain4j.service.Result;
+import dev.langchain4j.service.tool.ToolExecutor;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.core.utils.ListUtils;
+import io.kestra.plugin.ai.AIUtils;
+import io.kestra.plugin.ai.domain.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@Plugin(
+    examples =  {
+        @Example(
+            title = "Call a AI agent as a tool",
+            full = true,
+            code = {
+                """
+                    id: ai-agent-with-agent-tools
+                    namespace: company.ai
+
+                    inputs:
+                      - id: prompt
+                        type: STRING
+                        defaults: |
+                          Each flow can produce outputs that can be consumed by other flows. This is a list property, so that your flow can produce as many outputs as you need.
+                          Each output needs to have an id (the name of the output), a type (the same types you know from inputs e.g. STRING, URI or JSON) and value which is the actual output value that will be stored in internal storage and passed to other flows when needed.
+                    tasks:
+                      - id: ai-agent
+                        type: io.kestra.plugin.ai.agent.AIAgent
+                        provider:
+                          type: io.kestra.plugin.ai.provider.GoogleGemini
+                          modelName: gemini-2.5-flash
+                          apiKey: "{{ kv('GEMINI_API_KEY') }}"
+                        systemMessage: You are an AI agent, summarize the user message then translate it in french using the provided tool.
+                        prompt: "{{inputs.prompt}}"
+                        tools:
+                          - type: io.kestra.plugin.ai.tool.Agent
+                            description: An AI agent expert on translation
```

Comment:
```suggestion
                            description: Translation expert
```
what role does description play here? Is it meant for tool discovery?